edb/server/protocol/auth_ext/github.py

```diff
@@ -17,29 +17,28 @@
 #
 
 
+import http
 import httpx
 import urllib.parse
 
 from . import base
 
 
-class GitHubProvider(base.BaseProvider):
+class GitHubProvider(base.BaseOAuthProvider):
     def __init__(self, *args, **kwargs):
-        super().__init__("github", *args, **kwargs)
+        super().__init__("github", *args[1:], **kwargs)
 
-    def get_code_url(
-        self, state: str, redirect_uri: str, scope: str = "read:user"
-    ) -> str:
+    def _get_code_url(self, state: str, redirect_uri: str) -> str:
         params = {
             "client_id": self.client_id,
-            "scope": scope,
+            "scope": "read:user",
```

Comment:
I don't think the caller should ever need to customize this scope behavior: it feels firmly like something the Provider should know: how to get to the user info.